src/comps/main/comments/comments.jsx

```diff
@@ -1,37 +1,50 @@
-import React, { useState, useEffect } from "react";
-import comments from "./comment.css";
-import data from "../../../data/data.json";
+import React, { Component } from "react";
 import { allComments } from "../../../data/UserFunctions";
-export const Comments = props => {
-  const [commentList, setCommentList] = useState([]);
-
-  const activeId = props.activeVideoId;
-
-  function getComments() {
-    console.log("in");
+class CommentListComponent extends Component {
+  _isMounted = false;
+  constructor(props) {
+    super(props);
+    this.state = {
+      activeVideo: this.props.activeId,
+      commentList: []
+    };
+  }
+  componentDidMount() {
+    this._isMounted = true;
     allComments()
-      .then(res => setCommentList(res.data))
-      .catch(err => console.log(err));
+      .then(res => this.setState({ commentList: res.data }))
+      .catch(err => err);
   }
 
-  useEffect(() => {
-    getComments();
-  }, []);
-  const filter = commentList.filter(num => num.video_id == activeId);
-  // console.log(filter);
-  return (
-    <React.Fragment>
-      {filter.map((_data, index) => {
-        return (
-          <div className="card w-50 mt-3" key={_data.id}>
+  componentDidUpdate(prevProps) {
+    if (this.props !== prevProps) {
+      this.setState({
+        activeVideo: this.props.activeId,
+        commentList: this.props._arrComment
+      });
+    }
+  }
+
+  render() {
+    const renderComments = () => {
+      return this.state.commentList
+        .filter(comments => comments.video_id === this.state.activeVideo)
```

Comment:
Wait are you loading allllll the comments from the database and then filtering them every render?

Reply:
I fixed that 😅 my bad